modules/shadertools/src/lib/shader-module/shader-module.ts

```diff
@@ -43,6 +44,7 @@ export type ShaderModule<
 
   /** uniform buffers, textures, samplers, storage, ... */
   bindings?: Record<keyof BindingsT, {location: number; type: 'texture' | 'sampler' | 'uniforms'}>;
+  getBindings?: (settings?: any, prevBindings?: any) => Record<string, Texture | Sampler>;
```

Comment:
nit: Types could be a little more strict here I think, with `Record<keyof BindingsT, ...>`? Should `prevBindings` have the same type?

Reply:
+1